lib/Path/Tiny.pm

```diff
@@ -2415,6 +2415,50 @@ sub volume {
     return $self->[VOL];
 }
 
+=method change_extension
+
+    my $foo = path('C:/mydir/myfile.com.extension');
+    my $renamed_foo = $foo->change_extension('.old');
+
+Changes the extension of a path.
+
+The argument is the new extension (with or without a leading period).
+Specify undef to remove an existing extension from the path.
+
+Returns a Path::Tiny object.
+
+If extension is C<undef>, the returned string contains the specified path with its extension removed.
+If path has no extension, and extension is not C<undef>, the returned path string contains extension appended to the end of path.
+
+Current API available since 0.148.
+
+=cut
+
+sub change_extension {
+    my $self = shift;
+    my $new_extension = shift; # may be undef
+    
+    my $path_str = $self->stringify;
+    $path_str =~ s/\.[^.\/]+$//;  # Remove existing extension
+    
+    # If extension is undef, the returned string contains the specified path with its extension removed.
+    return path($path_str) unless defined $new_extension;
```

Comment:
I had exactly the same thoughts when implementing the feature but I didn't have additional opinions on this matter. 
The reason why I did it in the way as proposed was somewhat random. It seemed to be closer to the way how it is done in C#. However, the Path class in C# is a static class and cannot be instantiated at all - unlike Path::Tiny. 

I would like to propose that I change the implementation so that it mutates the calling path. Its imply feels better. In addition, the verb "change" when called as an objetc method IMO somewhat imples that it changes the object.